src/components/toast/toast.component.js

```diff
@@ -75,12 +82,22 @@ const Toast = React.forwardRef(
       if (onDismiss && !disableAutoFocus) {
```

Comment:
I think this can be condensed to 

```
     if (!disableAutoFocus) {
        if (open) {
          focusedElementBeforeOpening.current = document.activeElement;
          toastContentNodeRef.current?.focus();
          setHasBeenFocused(true);
        } else if (onDismiss && focusedElementBeforeOpening.current) {
          focusedElementBeforeOpening.current.focus();
          focusedElementBeforeOpening.current = undefined;
        }
     }
 ```

Reply:
But maybe we don't actually want it to focus without the `onDismiss` at all 🤔

Reply:
no, I think we do want it to focus even if there's no `onDismiss` unless of course `disableAutoFocus` is set. I put much of this original code in in order to focus the close button, and the only reason to check for `onDismiss` there was because that prop is (somewhat confusingly) what controls the presence or absence of the close button. I'm inclined to think, now we're going to focus the container, it should focus regardless of dismiss button or not.

Mind you if it times out then I suspect focus will be lost entirely when it disappears, which isn't great. Might be another question for UX... (I discussed some of this with Harpal before my changes, albeit mostly around how Toast should interact with a Dialog if launched from there)

Reply:
Cheers for the bits of context @robinzigmond 
I'll ask UX about the timeout thing too and we'll see what they come back regarding how focus should behave.